src/check-pkgs.js

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+const openwhisk = require('openwhisk');
+/**
+ * @param auth {string} openwhisk authentication key
+ * @param host {string} openwhisk api host
+ * @param namespace {string} openwhisk namespace under which actions exist
+ */
+function check_pkgs(auth, host, namespace, config) {
+  return new Promise((resolve, reject) => {
+    const ow = openwhisk({
+      api_key: auth,
+      apihost: host,
+      namespace,
+    });
+    ow.packages.list().then((data) => {
```

Comment:
why use promises and not async/await ?

Reply:
I wanted to have it resolved by the call to Promise.all

Reply:
I don't think it'll lead to a race condition, I checked and check-pkgs always resolves first. so it'll cause the entire Promise.all to fail before anything is written to the VCL/dictionaries, if you think it's not safe for racing reasons, I can move it. and use the await notation as you advise to ensure it executes first.

Reply:
I don't understand...`Promise.all()` doesn't guarantee the order of when the promises are resolved. however, this is not the point.

the `new Promise()` here is returned.... this is the same as writing:

```js
async function check_pkgs(auth, host, namespace, config) {
...
```

Reply:
arghhh duhhh it always returns a promise! I will fix this.